src/serialization/container.h

```diff
@@ -104,7 +123,8 @@ bool do_serialize_container(Archive<true> &ar, C &v)
       return false;
     if (i != v.begin())
       ar.delimit_array();
-    if(!::serialization::detail::serialize_container_element(ar, (typename C::value_type&)*i))
+    using serializable_value_type = typename ::serialization::detail::serializable_value_type<C>::type;
```

Comment:
I don't see why this needs to be cast, could you just pass `*i` to `serialize_container_element`? I suppose there are situations where `value_type`/`mapped_type` are not the same as `decltype(*i)`?

Reply:
We can't serialize `const` types without a lot of refactoring. The value type of mapped containers is `std::pair<const K, V>`, so when we try to serialize that value type, the pair serialization code tries to serialize `const K`, and the compiler can't find any functions that match `do_serialize(Archive&, const K&)` (since they don't exist) so it throws an error.

Reply:
That's too bad.